[READY] [FREDDY] [BETTY SPAGHETTI] Adds suicide to toy singularity

trollbreeder opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a suicide to the toy singularity, where you eat it and then you choke on it and die. Should've read the safety label.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
I tried suiciding with the toy singularity from Citrus the cargo sloth and it resorted to the default suicide, ruining my immersion. This rectifies that.

## Checklist
- [x] The code compiles.
- [x] The code is tested.
- [x] The code works as intended, with no major bugs found.
- [x] The code doesn't runtime.

## Changelog
:cl:
add: Nanotrasen reminds it's employees that consumption of any toys produced by it, and it's subsidiaries may result in choking or asphyxiation.
add: In other news, the Toy Singularity Eating Challenge is hot on the rise! Challenge your friends!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

trollbreeder committed:
Toy singularity suicide

I regret nothing.

trollbreeder committed:
tweaks to grammar

trollbreeder committed:
Changes playsound to be at user, not at src

The suicider is eating the toy, so it comes from the suicider, not the toy

trollbreeder committed:
Changes they and have to pronoun helpers

But he has no mouth!
But she has no mouth!
But it has no mouth!
But they have no mouth!

trollbreeder committed:
I forgot ()

GravitronBlue commented:
But what about if it's deep fried?

trollbreeder commented:
> But what about if it's deep fried?

Then it no longer does the unique suicide, since it's a different object (i presume it's obj/item/reagent_containers/food/snacks/deepfryholder)

GravitronBlue commented:
Point I meant to make was that trying to cunsume the toy results in death, deep frying it basically lets you eat it safely instead of choking.

This might just be a broader complaint since this basically applies to other deep fried items that sound like they're still far from being 'safe to eat.' (Deep fried glass shards, yummy)

Can probably just ignore my complaint due to only being about immersion.

GunHog commented:
This could be a generic toy suicide, the singularity deserves something unique, such as being compressed or consumed by it!

Shadowflame909 commented:
Aw thought this was gonna be a gib

trollbreeder commented:
I went for gib first but thought that'd be too harsh since we already have plenty of gib suicides (airtanks being the most obvious)

ATH1909 commented on code/game/objects/items/toys.dm:
```diff
@@ -178,6 +178,16 @@
+		user.visible_message("<span class='suicide'>[user] tries consuming [src]... but [user.p_they()] [user.p_have()] no mouth!") // and i must scream
+		return SHAME
+	user.visible_message("<span class='suicide'>[user] consumes [src]! It looks like [user.p_theyre()] trying to commit suicicide!</span>")
+	playsound(user, 'sound/items/eatfood.ogg', 50, TRUE)
```
maybe also play a choking or gasping sound? wasn't there a PR a while back that added those (that might have then gotten reverted?)?

trollbreeder replied:
It got reverted because it was too obnoxious.
#47287

ATH1909 commented:
> This could be a generic toy suicide, the singularity deserves something unique, such as being compressed or consumed by it!

Yeah, this should be a generic toy suicide, with the singularity toy, say, shrinking your character's size repeatedly until they disappear into nothingness.

JJRcop commented:
It's actually funnier that it doesn't do anything special and you choke on it.

dablusniper commented:
#50069

lordpidey commented on code/game/objects/items/toys.dm:
```diff
@@ -178,6 +178,16 @@
+		return SHAME
+	user.visible_message("<span class='suicide'>[user] consumes [src]! It looks like [user.p_theyre()] trying to commit suicicide!</span>")
+	playsound(user, 'sound/items/eatfood.ogg', 50, TRUE)
+	qdel(src)
```
Instead of Qdeleting it, could you implant it instead?

trollbreeder replied:
one second while i dig into the code

trollbreeder replied:
done, enjoy

trollbreeder committed:
more updates

fixes missing span
singulo now cavity implants the singulo into you
adds new shame check for if you already have a cavity in you while eating singulo
singulos now provide small amounts of nutrition

trollbreeder commented:
testing new code

trollbreeder committed:
it takes 3 seconds for you to choke to death

trollbreeder commented:
Now the toy singularity gets cavity implanted into you.
But if you already have a cavity implant, you regurgitate it and vomit it out.

lordpidey commented on code/game/objects/items/toys.dm:
```diff
@@ -178,6 +178,27 @@
+		user.visible_message("<span class='danger'>[user] regurgitates [src]!</span>") // I swear i dont have a fetish
+		user.vomit(100, TRUE, distance = 0)
+		user.adjustOxyLoss(120)
+		user.dropItemToGround(src) // incase the crit state doesn't drop the singulo to the floor
```
Honestly, I kinda want to chuck it across the room at supersonic speeds.

trollbreeder replied:
do you want distance to be random between 0 and 15?

lordpidey replied:
Sounds good.  Might also want to set vomit distance to the same amount.

trollbreeder replied:
on second thought, i'm not sure if this fits

actioninja commented on code/game/objects/items/toys.dm:
```diff
@@ -178,6 +178,27 @@
+	user.visible_message("<span class='suicide'>[user] consumes [src]! It looks like [user.p_theyre()] trying to commit suicicide!</span>")
+	playsound(user, 'sound/items/eatfood.ogg', 50, TRUE)
+	user.adjust_nutrition(50) // mmmm delicious
+	sleep(3 SECONDS)
```
don't use sleep, use timers

trollbreeder replied:
Saw sleep on fedora suicide and thought doing it here was okay, how would i go about doing this?

trollbreeder replied:
hold on i think i know what i can do

trollbreeder replied:
aight, sleep is changed to a callback timer

trollbreeder committed:
Changes sleep to timer, adds another failstate

A huge workaround just to avoid the demonized sleep() proc. Ah well.
Also if you throw the singulo away you choke to death

There might be issues with the regurgitating part of the suicide and the health scanner hud, but that's out of scope for this PR.

trollbreeder committed:
oops

documentation fail

trollbreeder commented:
travis documentation fail, apparantly

trollbreeder committed:
this should fix documentation

TiviPlus commented on code/game/objects/items/toys.dm:
```diff
@@ -178,6 +178,51 @@
+  * Throwing the singularity away will cause the user to start choking themself to death.
+  *
+  * user - Whoever is doing the suiciding
+  */
```
Proper format please:
  * Explanation
  *
  * Arguments:
  * * arg1 - Relevance of this argument

trollbreeder replied:
done

trollbreeder committed:
update documentation on this internal proc

ExcessiveUseOfCobblestone commented on code/game/objects/items/toys.dm:
```diff
@@ -178,6 +178,51 @@
+	user.visible_message("<span class='suicide'>[user] consumes [src]! It looks like [user.p_theyre()] trying to commit suicicide!</span>")
+	playsound(user, 'sound/items/eatfood.ogg', 50, TRUE)
+	user.adjust_nutrition(50) // mmmm delicious
+	addtimer(CALLBACK(src, .proc/manual_suicide, user), 30)
```
```suggestion
	addtimer(CALLBACK(src, .proc/manual_suicide, user), (3SECONDS))
```

trollbreeder replied:
Already commited by cobby

ExcessiveUseOfCobblestone committed:
Update code/game/objects/items/toys.dm
